optuna/storages/rdb/storage.py

```diff
@@ -437,7 +437,13 @@ def set_trial_system_attr(self, trial_id, key, value):
         session = self.scoped_session()
 
         trial = models.TrialModel.find_or_raise_by_id(trial_id, session)
-        self.check_trial_is_updatable(trial_id, trial.state)
+        if key == '_number':
+            # If this trial was created before v0.9.0, it doesn't have `_number` attribute.
+            # We want to set the attribute regardless the trial state,
```

Comment:
In my opinion, we've better add a bit more pieces of information about this exceptional logic.

For example:
```
`_number` attribute may be set even after a trial is finished.
This happens if the trial was created before v0.9.0, where a trial didn’t have `_number` attribute.
In this case, `check_trial_is_updatable` is skipped to avoid the `RuntimeError`.
```

Reply:
Sounds good. I have updated the comment at https://github.com/pfnet/optuna/pull/415/commits/321931adcc6f478df39413e2e5c5543039ca56a1.